media-libs/ctl/files/ctl-1.5.2-Use-GNUInstallDirs-and-fix-PkgConfig-files.patch

```diff
@@ -0,0 +1,371 @@
+From e284bd943f7c72b13c4b2bd508227ba3657ff6ca Mon Sep 17 00:00:00 2001
+From: Jonathan Scruggs <j.scruggs@gmail.com>
+Date: Mon, 25 Sep 2017 17:41:26 +0100
+Subject: [PATCH] CTL: Use GNUInstallDirs and fix PkgConfig files
+
+---
+ CMakeLists.txt                         | 36 +++++++++-------------------------
+ OpenEXR_CTL/CMakeLists.txt             |  2 +-
+ OpenEXR_CTL/CtlModules/CMakeLists.txt  |  2 +-
+ OpenEXR_CTL/exr_ctl_exr/CMakeLists.txt |  4 ++--
+ OpenEXR_CTL/exrdpx/CMakeLists.txt      |  4 ++--
+ config/CTL.pc.in                       |  8 +++-----
+ config/OpenEXR_CTL.pc.in               | 15 +++++++-------
+ configure.cmake                        | 32 +++++++++---------------------
+ ctlrender/CMakeLists.txt               | 12 +-----------
+ doc/CMakeLists.txt                     |  2 +-
+ lib/IlmCtl/CMakeLists.txt              |  4 ++--
+ lib/IlmCtlMath/CMakeLists.txt          |  4 ++--
+ lib/IlmCtlSimd/CMakeLists.txt          |  6 +++---
+ lib/IlmImfCtl/CMakeLists.txt           |  6 +++---
+ lib/dpx/CMakeLists.txt                 |  1 +
+ 15 files changed, 47 insertions(+), 91 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 01f622c..35e9a48 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,22 +1,14 @@
+ cmake_minimum_required(VERSION 2.8)
+ project( CTL )
+ 
++INCLUDE ( GNUInstallDirs )
++
+ set( CTL_MAJOR_VERSION 1 )
+ set( CTL_MINOR_VERSION 5 )
+-set( CTL_PATCH_VERSION 0 )
++set( CTL_PATCH_VERSION 2 )
+ set( CTL_VERSION ${CTL_MAJOR_VERSION}.${CTL_MINOR_VERSION}.${CTL_PATCH_VERSION} )
+ 
+-## Make install directories overrideable
+-set( INSTALL_LIB_DIR lib CACHE PATH "Install directory for libraries" )
+-set( INSTALL_BIN_DIR bin CACHE PATH "Install directory for executable binaries" )
+-set( INSTALL_INCLUDE_DIR include CACHE PATH "Install directory for public header files" )
+-set( INSTALL_DOC_DIR doc CACHE PATH "Install directory for documentation" )
```

Comment:
No. Can't keep those otherwise things will install to lib, which fails multilib-strict, so it must be replaced by GNUInstallDirs which will choose lib64 or lib depending on the system. Even if Upstream doesn't accept the patch, Gentoo will always need to patch those directories out.

Reply:
it looks to me -DINSTALL_LIB_DIR=lib64 will install to lib64 ?

Reply:
I see your point about not changing the variables and just using -DINSTALL_LIB_DIR=${get_libdir}. However, that would not work so well for cross-compiling and prefix systems. GNUInstallDirs is the proper fix to this issue. Granted this patch isn't as complete as it could be since there is a lot to be removed still, like the find IlmBase and OpenEXR cmake files and serious space issues, but that makes the patch well over 20kb. But this patch in this simple form is the best way to make this proper -- especially for cross-compiling and having a prefix. Gentoo's cmake eclasses play nice with GNUInstallDirs and do things **automatically** for us. This patch also makes CTL compatible with old and current versions of IlmBase/OpenEXR and possible future ones (I need to try the latest trunk, but the logic is there to support it. So, CTL 1.5.2 can eventually be made stable and old versions removed along with media-libs/openexr_ctl being last rited and removed.

Also, to be honest, I'm not really interested in CTL and OpenEXR_Viewers, just IlmBase and OpenEXR as they affect the packages that I maintain. Can I ask that at the very least, IlmBase and OpenEXR get reviewed and compile tested and checked to be included into the main repo so I can make PRs for OpenImageIo 1.7, which OSL and the new Blender 2.79 depend on? Then CTL can be figured out? I can split out IlmBase and OpenEXR to separate PRs if needed.

Reply:
I agree GNUInstallDirs is better; I'm not so convinced it helps cross compiling or prefix but being more standard it certainly helps simplifying and avoiding mistakes. What I'm worried about is if upstream will think the same and if they did not purposedly avoid it for a reason unknown to me. 

That's why, while submitting your PR upstream is absolutely great, I'm not certain if it is not still simpler for us to simply feed cmake the variables upstream uses and avoid patching.

Anyway, I think this PR is good to go now, lemme do final checks and merge if ok

Reply:
All valid points.  I really hope they accept upstream as then they don't need checks if they are on Windows, MacOS, or Linux, so it really saves a lot of code and makes things worry free. :)